cmd/neofs-node/object.go

```diff
@@ -664,20 +661,30 @@ func (o objectSource) Head(ctx context.Context, addr oid.Address) (*objectSDK.Ob
 	return hw.h, err
 }
 
-func (o objectSource) Search(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters) ([]oid.ID, error) {
-	var sw searchWriter
-
-	var sPrm searchsvc.Prm
-	sPrm.SetWriter(&sw)
-	sPrm.WithSearchFilters(filters)
-	sPrm.WithContainerID(cnr)
-
-	err := o.search.Search(ctx, sPrm)
+func (o objectSource) SearchOne(ctx context.Context, cnr cid.ID, filters objectSDK.SearchFilters) ([]oid.ID, error) {
```

Comment:
If it's `SearchOne` it can just return `oid.ID` and caller can check for `.IsZero()`.